src/librbd/api/Mirror.cc

```diff
@@ -3083,43 +3035,27 @@ int Mirror<I>::group_promote(IoCtx& group_ioctx, const char *group_name,
     return r;
   }
 
+  mirror::PromotionState promotion_state;
   cls::rbd::MirrorGroup mirror_group;
-  r = cls_client::mirror_group_get(&group_ioctx, group_id, &mirror_group);
+  C_SaferCond ctx;
+  // pass group_id to get ENOENT error returned back when group is not mirror enabled
+  auto request = mirror::GroupGetInfoRequest<I>::create(
+    group_ioctx, "", group_id, &mirror_group, &promotion_state, &ctx);
+  request->send();
+  r = ctx.wait();
   if (r == -ENOENT) {
     ldout(cct, 10) << "mirroring for group " << group_name
-                   << " disabled" << dendl;
-    return -EINVAL;
+                   << " already disabled" << dendl;
```

Comment:
"mirroring for group already disabled" makes it sound like the command was trying to disable mirroring on the group.
The original message was more appropriate. Alternate suggestion:

"group is not enabled for mirroring"